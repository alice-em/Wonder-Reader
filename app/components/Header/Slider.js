import React, { Fragment, memo, useEffect } from 'react';

import Context from '../../context/ContextFactory';
import SliderInput from './SliderInput';

const boxShadow = [
  'inset rgb(135, 169, 214) 0px 3px 0px',
  'inset rgba(0, 0, 0, 0.15) 0px 10px 10px',
].join(', ');

const styles = {
  Slider: {
    border: '1px solid rgba(255,255,255,0.3)',
    borderRadius: '5px',
    borderTop: '2px solid rgba(255,255,255,0.8)',
    boxShadow,
    display: 'flex',
    float: 'left',
    marginTop: '7px',
    padding: '3px',
  },
  wide: {
    width: '100px',
  },
  zoomLevel: {
    cursor: 'default',
    fontFamily: 'Carter One',
    fontSize: '20px',
    marginRight: '5px',
    textAlign: 'right',
    width: '45px',
  },
};

const sliderComponent = document.getElementById('sliderComponent');
const blurSliderInput = () => {
  document.getElementById('SliderInput').blur();
};

const Slider = () => {
  useEffect(() => {
    if (sliderComponent) {
      sliderComponent.addEventListener('mouseleave', blurSliderInput);
    }

    return () => {
      if (sliderComponent) {
        sliderComponent.removeEventListener('mouseleave', blurSliderInput);
      }
    };
  }, []);

  return (
    <div className="slider" id="sliderComponent" style={styles.Slider}>
      <Context.Consumer>
        {({ state: { zoomLevel } }) => (
          <Fragment>
            <SliderInput value={zoomLevel} />
            <div className="zoomLevel" style={styles.zoomLevel}>
              {zoomLevel}
            </div>
          </Fragment>
        )}
      </Context.Consumer>
    </div>
  );
};

export { Slider };
export default memo(Slider);
