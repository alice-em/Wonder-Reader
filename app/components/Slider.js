import React, { Component } from 'react';

import ZoomDisplay from './ZoomDisplay';
import ZoomInput from './ZoomInput';

import * as actions from '../actions';
import * as store from '../store';

// const sliderComponent = document.getElementById('sliderComponent');
const sliderInput = document.getElementById('sliderInput');

const boxShadow = (
  'inset rgb(135, 169, 214) 0px 3px 0px,'
  + 'inset rgba(0, 0, 0, 0.15) 0px 10px 10px'
);

const styles = {
  alignItems: 'center',
  border: '1px solid rgba(255,255,255,0.3)',
  borderRadius: '5px',
  borderTop: '2px solid rgba(255,255,255,0.8)',
  boxShadow,
  display: 'flex',
  justifyContent: 'center',
  marginTop: '7px',
  padding: '3px'
};

class Slider extends Component {
  state = {
    zoomLevel: store.zoom.getZoomLevel()
  };

  componentDidMount() {
    store.zoom.on('change', this.setZoomLevelState);
  }

  componentWillUnmount() {
    store.zoom.removeListener('change', this.setZoomLevelState);
  }

  blurSliderInput = () => {
    sliderInput.blur();
  };

  onChange = e => {
    const zoomLevel = Number(e.target.value);
    actions.zoom.setZoomLevel(zoomLevel);
  };

  setZoomLevelState = () => {
    this.setState({
      zoomLevel: store.zoom.getZoomLevel()
    });
  };

  render() {
    const { zoomLevel } = this.state;
    return (
      <div
        className="slider"
        id="sliderComponent"
        onBlur={this.blurSliderInput}
        style={styles}
      >
        <ZoomInput
          onChange={this.onChange}
          value={zoomLevel}
        />
        <ZoomDisplay value={zoomLevel} />
      </div>
    );
  }
}

export default Slider;
