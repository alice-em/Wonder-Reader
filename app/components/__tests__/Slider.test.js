import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

import Slider from '../Slider';

Enzyme.configure({ adapter: new Adapter() });

describe('Slider', () => {
  const onChangeFunc = jest.fn();
  it('should render', () => {
    const wrapper = shallow(<Slider onChange={onChangeFunc} value={100} />);
    expect(wrapper).toMatchSnapshot();
  });
});
