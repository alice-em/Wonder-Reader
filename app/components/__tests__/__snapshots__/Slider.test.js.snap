// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slider
    onChange={[MockFunction]}
    value={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SliderInput
          onChange={[Function]}
          value={100}
        />,
        <div
          className="zoomLevel"
          style={
            Object {
              "cursor": "default",
              "fontFamily": "Carter One",
              "fontSize": "20px",
              "marginRight": "5px",
              "textAlign": "right",
              "width": "45px",
            }
          }
        >
          100
        </div>,
      ],
      "className": "slider",
      "id": "sliderComponent",
      "style": Object {
        "border": "1px solid rgba(255,255,255,0.3)",
        "borderRadius": "5px",
        "borderTop": "2px solid rgba(255,255,255,0.8)",
        "boxShadow": "inset rgb(135, 169, 214) 0px 3px 0px, inset rgba(0, 0, 0, 0.15) 0px 10px 10px",
        "display": "flex",
        "float": "left",
        "marginTop": "7px",
        "padding": "3px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChange": [Function],
          "value": 100,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 100,
          "className": "zoomLevel",
          "style": Object {
            "cursor": "default",
            "fontFamily": "Carter One",
            "fontSize": "20px",
            "marginRight": "5px",
            "textAlign": "right",
            "width": "45px",
          },
        },
        "ref": null,
        "rendered": 100,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SliderInput
            onChange={[Function]}
            value={100}
          />,
          <div
            className="zoomLevel"
            style={
              Object {
                "cursor": "default",
                "fontFamily": "Carter One",
                "fontSize": "20px",
                "marginRight": "5px",
                "textAlign": "right",
                "width": "45px",
              }
            }
          >
            100
          </div>,
        ],
        "className": "slider",
        "id": "sliderComponent",
        "style": Object {
          "border": "1px solid rgba(255,255,255,0.3)",
          "borderRadius": "5px",
          "borderTop": "2px solid rgba(255,255,255,0.8)",
          "boxShadow": "inset rgb(135, 169, 214) 0px 3px 0px, inset rgba(0, 0, 0, 0.15) 0px 10px 10px",
          "display": "flex",
          "float": "left",
          "marginTop": "7px",
          "padding": "3px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChange": [Function],
            "value": 100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 100,
            "className": "zoomLevel",
            "style": Object {
              "cursor": "default",
              "fontFamily": "Carter One",
              "fontSize": "20px",
              "marginRight": "5px",
              "textAlign": "right",
              "width": "45px",
            },
          },
          "ref": null,
          "rendered": 100,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
